app/controllers/authentication_controller.rb

```diff
@@ -9,7 +9,17 @@ def self.call(env)
 
   def redirect_to_omniauth
     store_location(attempted_path) if request.get?
-    redirect_to "/auth/#{default_provider}"
+    if default_provider == "auth0" && e2e_user?
+      # the value passed in connection must match the name given in the auth0 terraform
+      redirect_to "/auth/#{default_provider}?connection=Username-Password-Authentication"
+    else
+      redirect_to "/auth/#{default_provider}"
+    end
```

Comment:
`redirect_to` doesn't stop the execution of a function [[1](https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_to)], so the way this is written makes it easier for us to accidentally do a double redirect in future.

What about building up a params variable and then constructing a URL before handing off to `redirect_to`?

```suggestion
    params = {}
    if default_provider == "auth0" && e2e_user?
      # the value passed in connection must match the name given in the auth0 terraform
      params[:connection] = Username-Password-Authentication"
    end
    
    redirect_to ActionDispatch::Http::URL.path_for(path: "/auth/#{default_provider}", params:)
```

Reply:
I've made this change now, thanks for the suggestion!